src/Pages/Quiz/quiz.jsx

```diff
@@ -0,0 +1,134 @@
+import { useReducer, useState } from "react";
+import { Link } from "react-router-dom";
+
+// Components
+import { Rules } from "./rules";
+
+// data
+import data from "../../data";
+
+function Quiz() {
+  // useState to change background of selected option
+  const [bg, setBg] = useState({});
+
+  // reducer function to manipulate state according to action type i.e next and prev buttons
+  function reducerFn(state, action) {
+    switch (action.type) {
+      // case to move to prev ques
+      case "PrevQues":
+        return {
+          ...state,
+          index: state.index === 0 ? state.index : state.index - 1,
+        };
+      // case to go to next ques
+      case "NextQues":
+        return {
+          ...state,
+          index:
+            state.index === data.results.length - 1
+              ? state.index
+              : state.index + 1,
+        };
+      default:
+        return state;
+    }
+  }
+
+  // Function to change the background color of selected option
+  function optionHandler() {
+    setBg({
+      backgroundColor: "var(--PRIMARY-COLOR)",
+      color: "var(--SECONDARY-COLOR)",
+    });
+  }
+
+  // useReducer
+  const [state, dispatch] = useReducer(reducerFn, {
+    index: 0,
+  });
```

Comment:
useReducer should be used when you've multiple states. 
Here since you've less than 3 states so useState is more appropriate here.

Reply:
I will be adding more states for the quiz.